express-notes-api/index.js

```diff
@@ -0,0 +1,113 @@
+const express = require("express");
+const app = express();
+const fs = require("fs");
+const notesJson = require("./data.json");
+
+const writingFile = () => {};
+
+app.use(express.json());
+
+app.get("/api/notes", (req, res) => {
+  const gradesArray = [];
+  if (notesJson.notes) {
+    for (let notes in notesJson.notes) {
+      gradesArray.push(notesJson.notes[notes]);
+    }
+  } else if (!notesJson.notes) {
+    res.json([]);
+  }
+  res.json(gradesArray);
+});
+
+app.get("/api/notes/:id", (req, res) => {
+  let noteGet = {};
+
+  for (let notes in notesJson.notes) {
+    if (notesJson.notes[req.params.id]) {
+      noteGet = notesJson.notes[req.params.id];
+    } else if (req.params.id < 0) {
+      noteGet.error = "Id must be a positive integer";
+      res.status(400);
+    } else if (!notesJson.notes[req.params.id]) {
+      noteGet.error = `Error: cannot find note with id ${req.params.id}`;
+      res.status(404);
+    }
+  }
+
+  res.json(noteGet);
+});
+
+app.post("/api/notes", (req, res) => {
+  const content = req.body.content;
+  let newNote = {};
+  if (content === undefined) {
+    newNote.error = "content is a require field";
+    res.status(400).json(newNote);
+  } else if (content) {
+    const id = notesJson.nextId++;
+    newNote = {
+      id,
+      content,
+    };
+    notesJson.notes[id] = newNote;
+    fs.writeFile(`./data.json`, JSON.stringify(notesJson), (err) => {
+      if (err) {
+        console.error("An unexpected error occurred");
+        res.status(500).json({ error: "An unexpected error occurred." });
+      } else {
+        res.status(201).json(newNote);
+      }
+    });
+  }
+});
+
+app.delete("/api/notes/:id", (req, res) => {
+  let deletedNote = {};
+  const id = req.params.id;
+  if (id < 0) {
+    deletedNote.error = "id must be a positive integer";
+    res.status(400);
```

Comment:
A `400 Bad Request` should be produced for _any_ `id` which isn't a positive integer:

---

![image](https://user-images.githubusercontent.com/16143771/144653724-9df5319a-d6ed-4030-8013-9d5806373acd.png)

---

![image](https://user-images.githubusercontent.com/16143771/144653791-1d4a959a-59ee-4769-b76c-75ccd684a732.png)